packages/prosemirror-lwdita-demo/src/demo-plugin.ts

```diff
@@ -152,9 +152,13 @@ function publishGithubDocument(config: Config, localization: Localization, urlPa
     if (dispatch) {
       dispatch(state.tr);
       const documentNode = transformToJditaDocumentNode(state);
+      // remove the base url from the documentNode
+      const baseUrl = urlParams.referer.split('/').slice(0, -1).join('/');
```

Comment:
The baseUrl should be the full document URL. I don't think we should be treating URLs as strings in this way as it is error prone. We should use the URL API to do the resolution, e.g. `new URL('images/my-image.png', baseUrl)` - see: https://developer.mozilla.org/en-US/docs/Web/API/URL/URL

Reply:
@adamretter I tried this and it does not produce a correct url:
```js
let baseUrl = "https://evolvedbinary.github.io/cityehr-documentation/verify-install.html"
let img = "/images/post-authentication-page.png"
new URL(img, baseUrl).href

// returns a link with just the origin and the image path
// https://evolvedbinary.github.io/images/post-authentication-page.png
// it should also include the relative path
// https://evolvedbinary.github.io/cityehr-documentation/images/post-authentication-page.png
```

I have opted for this solution instead 
```js
const rawDoc = await fetchRawDocumentFromGitHub(config, ghrepo, source, branch);
  // update the document with the relative path

  const updatedDoc = rawDoc.replace(/href="([^"]+)"/g, (match, url) => {
    // avoid changing the url if it is already an absolute url
    if(url.startsWith('http') || url.startsWith('https')) {
      return match;
    }
    // get the base url from the referer
    const baseUrl = referer.split('/').slice(0, -1).join('/');
    // return the updated href url
    return `href="${baseUrl}/${url}"`;
  });

  const jsonDoc = await transformGitHubDocumentToProsemirrorJson(updatedDoc);
```

Reply:
Fixed in https://github.com/evolvedbinary/prosemirror-lwdita/pull/479/commits/2ab2b7eda37f3205521c2df2f7d86c9407227173

Reply:
@marmoure  The URL API is behaving correctly - actually there is an issue in your example code! Instead of the URL that is relative to the root of the FQDN, i.e.:
```
let img = "/images/post-authentication-page.png"
```
you should instead have a URL that is relative to the document, e.g.:
```
let img = "images/post-authentication-page.png"
```
**NOTE**: the removed '/' at the start.